docs(config): notes vite.config replace __dirname/__filename (#6551)
vitejs/vite: docs/config/index.md

vite --config my-config.js
```

> Note Vite will replace `__filename`, `__dirname`, and `import.meta.url`. Using these as variable names will result in an error:

Reviewer: ```suggestion
Note that Vite will replace `__filename`, `__dirname`, and `import.meta.url`. Using these as variable names will result in an error:
```

vite --config my-config.js
```

> Note Vite will replace `__filename`/`__dirname`/`import.meta.url` with the correct values.
> 
> Using `__filename`/`__dirname` as variable names will be an error:

Reviewer: ```suggestion
Note Vite will replace `__filename`, `__dirname`, and `import.meta.url`. Using these as variable names will result in an error:
```

```js

> // example
> const __filename = "value"
> // will compile with
> const "path/vite.config.js" = "value"

Reviewer: ```suggestion
const __filename = "value"
// will be transformed to
const "path/vite.config.js" = "value"
```
